tools/box/box.h

```diff
@@ -53,38 +56,122 @@ jxl::Status ParseBoxHeader(const uint8_t** next_in, size_t* available_in,
 // TODO(lode): streaming C API
 jxl::Status AppendBoxHeader(const Box& box, jxl::PaddedBytes* out);
 
+/* Known/defined types:
+  "Exif"
+     The exif data has the format of 'Exif block' as defined in
+     ISO/IEC23008-12:2017 Clause A.2.1
+     Note: it starts with a tiff header offset of 4 bytes (usually zeroes),
+     and the actual Exif data (starting with the tiff header MM or II) is
+     located at that offset.
+     // TODO(lode): support the theoretical case of multiple exif boxes
+  "xml "
+     Contains XML data, typically used for XMP.
+  "jumb"
+     JUMBF superbox data.
+     The parsing of the nested boxes inside is not handled here.
+
+  Any of these boxes (and possibly others) can be stored in a compressed way
+  in a "brob" box, which starts with 4 uncompressed bytes specifying the
+  actual box type, followed by a Brotli stream.
+
+  To facilitate transparent handling of boxes, the BrobBlob struct is used.
+  It has pointers to both the uncompressed data and the compressed data.
+  When decoding:
+    - when reading, either udata* or cdata* will be set
+    - in the uncompressed case, only udata* is used.
+    - in the compressed case, cdata* points into the input stream, and
+      when accessing the box with getBlob(), pbytes will store the
+      decompressed data, and udata* will point to it.
+  When encoding:
+    - in the uncompressed case, only udata* is used.
+    - in the compressed case, udata* points to the uncompressed input, and
+      when adding the box with addBlob(), pbytes will store the compressed
+      data, and cdata* will point to it.
+    - when writing, a "brob" box will be written if cdata* is nonzero,
+      otherwise an uncompressed box will be written.
+*/
+struct BrobBlob {
+  uint8_t type[4];
+  // uncompressed data (mutable: decompressed data can be added or removed)
+  mutable const uint8_t* udata = nullptr;  // Not owned
```

Comment:
why is this mutable?

Reply:
Because I want to be able to take a const BrobBlob and add (or I guess remove) its alternative, redundant representation (decompress it if it was compressed, compress it if it was uncompressed), which does change the struct but it doesn't conceptually change its semantical value, so it should be something you can do to const BrobBlobs.

Probably there's a cleaner way to do this, and if not, then probably it's a better idea to make those mutable fields private.